### PR TITLE
[KBFS-1526] Plumb through ExtraMetadata some more

### PR DESCRIPTION
**Pull request by akalin-keybase**
I missed a few places in my last PR (#430).

Make ImmutableBareRootMetadata have an
ExtraMetadata field.

Add constructor for RootMetadata (to avoid
dropping fields).

Also add checks for wkbID and rkbID being either
both nil or both non-nil.

### CODE REVIEW

**Comment by akalin-keybase** [comment]
fyi @aalness

File: libkbfs/journal_md_ops.go [hunk 1]
```diff
@@ -152,7 +150,8 @@ func (j journalMDOps) getRangeFromJournal(
 
 	for _, ibrmd := range ibrmds {
 		irmd, err :=
-			tlfJournal.convertImmutableBareRMDToIRMD(ctx, ibrmd, handle)
+			tlfJournal.convertImmutableBareRMDToIRMD(
```
Comment (strib) [inline libkbfs/journal_md_ops.go]: And reason to add a line here?

Comment (akalin-keybase) [inline libkbfs/journal_md_ops.go]: 80 chars. But I got it to fit in two lines.

**Comment by akalin-keybase** [comment]
Fairly sure that CI failure is a flaky hang; filed https://keybase.atlassian.net/browse/KBFS-1641

**Pull request merged**